vmm/src/cpu.rs

```diff
@@ -919,12 +921,14 @@ impl CpuManager {
 
                                 #[cfg(feature = "kvm")]
                                 {
-                                    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(true);
-                                    if !matches!(vcpu.lock().unwrap().run(), Ok(VmExit::Ignore)) {
-                                        error!("Unexpected VM exit on \"immediate_exit\" run");
-                                        break;
+                                    if on_kvm {
```

Comment:
Out of curiosity, why is this needed given it's already behind a KVM feature gate?

Reply:
This issue was found when running on a dual binary (i.e. both kvm and mshv compiled in) running on mshv. `set_immediate_exit` is only implemented on kvm (and wouldn't be needed on mshv due to the different method of decoding mmio exits.)

Reply:
Gotcha!